src/models/miner.model.js

```diff
@@ -40,19 +63,142 @@ class MinerModel {
      */
     getJob() {
         return new Promise((resolve, reject) => {
+            
+
+
+
+
             resolve('Boilerplate for Miner::getJob');
         })
     }
-
     /**
-     * @param {BlockTemplate} blockTemplate 
-     */
-    submit(blockTemplate) {
+     * 
+     * @Example Request
+     * "id": 2,
+     * "jsonrpc": "2.0",
+     * "method": "submit",
+     * "params": {
+     * "id": "1be0b7b6-b15a-47be-a17d-46b2911cf7d0",
+     * "job_id": "4BiGm3/RgGQzgkTI/xV0smdA+EGZ",
+     * "nonce": "d0030040",
+     * "result": "e1364b8782719d7683e2ccd3d8f724bc59dfa780a9e960e7c0e0046acdb40100"
+     * }
+     * @param {request} req
+     */ 
+    submit(req) {
         return new Promise((resolve, reject) => {
-            resolve('Boilerplate for Miner::submit');
+            //miner, job, blockTemplate, nonce, resultHash
+            const minerData = 
+            {
+                "id": req.params.id,
+                "job_id": req.params.job_id,
+                "nonce": req.params.nonce,
+                "result": req.params.result
+            }
+            /*
+             minerdata.id = miner uuid
+             minerdata.job_id = job id
+             minerdata.nonce = nonce
+             minerdata.result = resultHash -> the hash of the supposed block
+             Reconstruct the block from data,check to see that it matches the sent block
+            */
+            
+            const blockTemplate = getCurrentBlockTemplate();
+            const job = getJobById();
+            
+            constructBlock = function(blockTemplate, minerData, job){
+            var block = new Buffer(blockTemplate.buffer.length);
+            blockTemplate.buffer.copy(block);
+            block.writeUInt32BE(job.extraNonce, blockTemplate.reserveOffset);
+            new Buffer(minerData.nonce, 'hex').copy(block, 39);
+            return block;
+            }
+
+            var block = constructBlock(blocktemplate,minerData, job);
+
+            const convertedBlob = cryptoNoteUtils.cnUtil.convert_blob(block);
+
+            const algoVariant = 0;
+            const blockHashed = cryptoNight(convertedBlob, algoVariant);
+            
+            if (blockHashed.toString('hex') !== minerdata.result) {
+                logger("Bad Hash, penalty to miner");
+                resolve("Bad Hash");
+                return false;
+            }
+        
+            var hashArray = blockHashed.toJSON().reverse();
+            // Diff is a reference from bignum 
+            var hashDiff = diff.div(bignum.fromBuffer(new Buffer(hashArray)));
+        
+            if (hashDiff.ge(blockTemplate.difficulty)){
+                moneroApi.submit(block, function(error, result){
```

Comment:
This also returns a promise instead of taking a callback
```suggestion
                return moneroApi.submit(block).then((...) => {...})
```